frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts

```diff
@@ -9,72 +9,248 @@ import {
   const application = applicationFixture(
     fixtures.enduserChildFixtureJari,
-    fixtures.enduserGuardianFixture
+    fixtures.enduserGuardianFixture,
+    undefined,
+    'preschool',
+    null,
+    [daycareFixture.id],
+    true
   )
   applicationId = application.id
   await insertApplications([application])
 
-  const decision = await Fixture.decision()
-    .with({
-      applicationId: application.id,
-      employeeId: employee.data.id,
-      unitId: application.form.apply.preferredUnits[0],
-      startDate: application.form.preferredStartDate
-    })
-    .save()
+  await execSimpleApplicationActions(applicationId, [
+    'move-to-waiting-placement',
+    'create-default-placement-plan',
+    'send-decisions-without-proposal'
+  ])
+  await runPendingAsyncJobs()
+
+  const decisions = await getDecisionsByApplication(applicationId)
+  if (decisions.length !== 2) throw Error('Expected 2 decisions')
+  const preschoolDecisionId = decisions.find((d) => d.type === 'PRESCHOOL')?.id
+  if (!preschoolDecisionId)
+    throw Error('Expected a decision with type PRESCHOOL')
+  const preschoolDaycareDecisionId = decisions.find(
+    (d) => d.type === 'PRESCHOOL_DAYCARE'
+  )?.id
+  if (!preschoolDaycareDecisionId)
+    throw Error('Expected a decision with type PRESCHOOL_DAYCARE')
 
   await t.useRole(enduserRole)
   await t.click(citizenHomePage.nav.decisions)
 
   await t
     .expect(citizenDecisionsPage.unresolvedDecisionsInfoBox.textContent)
-    .contains('1 päätös odottaa vahvistusta')
+    .contains('2 päätöstä odottaa vahvistusta')
+
+  await citizenDecisionsPage.assertApplicationDecision(
+    applicationId,
+    preschoolDecisionId,
+    `${enduserChildFixtureJari.firstName} ${enduserChildFixtureJari.lastName}`,
+    'Päätös esiopetuksesta',
+    format(new Date(application.form.preferredStartDate), 'dd.MM.yyyy'),
+    'Vahvistettavana huoltajalla'
+  )
 
   await citizenDecisionsPage.assertApplicationDecision(
     applicationId,
-    decision.data.id,
+    preschoolDaycareDecisionId,
     `${enduserChildFixtureJari.firstName} ${enduserChildFixtureJari.lastName}`,
-    'Päätös varhaiskasvatuksesta',
+    'Päätös liittyvästä varhaiskasvatuksesta',
     format(new Date(application.form.preferredStartDate), 'dd.MM.yyyy'),
     'Vahvistettavana huoltajalla'
   )
+
+  await t.click(citizenDecisionsPage.goRespondToDecisionBtn(applicationId))
+
+  // === Response page for decisions of a single application ===
+
+  await t
+    .expect(citizenDecisionResponsePage.pageTitle.textContent)
+    .contains('Päätökset')
+
+  await t
+    .expect(citizenDecisionResponsePage.unresolvedDecisionsInfoBox.textContent)
+    .contains('2 päätöstä odottaa vahvistusta')
+
+  // preschool_daycare decision cannot be accepted before accepting preschool
+  await t
+    .expect(
+      citizenDecisionResponsePage
+        .submitResponseBtn(preschoolDaycareDecisionId)
+        .hasAttribute('disabled')
+    )
+    .ok()
+
+  // accept preschool decision
```

Comment:
These commented blocks should go to the citizen-decision page object (tests in spec, page structure in page object)

Reply:
added 2 commits, now better?